firefox-ios/WidgetKit/TopSites/TopSitesWidget.swift

```diff
@@ -21,78 +21,67 @@ struct TopSitesWidget: Widget {
 }
 
 struct TopSitesView: View {
+    private struct UX {
+        static let widgetBackgroundColor = Color(red: 0.11, green: 0.11, blue: 0.13)
+        static let emptySquareFillColor = Color(red: 0.85, green: 0.85, blue: 0.85, opacity: 0.3)
+        static let maskShapeCornerRadius: CGFloat = 5.0
+    }
+
     let entry: TopSitesEntry
 
     @ViewBuilder
-    func topSitesItem(_ site: WidgetKitTopSiteModel) -> some View {
+    func topSitesItem(_ site: WidgetKitTopSiteModel, itemSize: CGFloat) -> some View {
         let url = site.url
 
         Link(destination: linkToContainingApp("?url=\(url)", query: "widget-medium-topsites-open-url")) {
-            if entry.favicons[site.imageKey] != nil {
-                (entry.favicons[site.imageKey])!.resizable().frame(width: 60, height: 60).mask(maskShape)
-            } else {
+            let imageSize = itemSize / 2
+            if let image = entry.favicons[site.imageKey] {
                 Rectangle()
-                    .fill(UIColor(red: 0.85, green: 0.85, blue: 0.85, alpha: 0.3).color)
-                    .frame(width: 60, height: 60)
+                    .fill(Color.clear)
+                    .overlay {
+                        image
+                            .resizable()
+                            .frame(width: imageSize, height: imageSize)
+                            .scaledToFit()
+                            .mask(maskShape)
+                    }
+                    .mask(maskShape)
+            } else {
+                emptySquare.frame(width: imageSize, height: imageSize)
             }
-        }
+        }.frame(width: itemSize, height: itemSize)
     }
```

Comment:
What do you think about making the TopSitesItem its own `View` class in a separate file? 🤔  Possibly EmptySquare as well.

I will let you decide if it's a good refactor.

Reply:
I removed the empty square in the end 😁. For the top item i wouldn't make a separate component unless we have to reuse it somewhere else.